hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/handle/record/SystemSetup.java

```diff
@@ -106,13 +122,95 @@ public void doGenesisSetup(@NonNull final Dispatch dispatch) {
+                        adminConfig.upgradeFeeSchedulesFile(),
+                        SystemSetup::parseFeeSchedules),
+                new AutoSysFileUpdate(
+                        createFileID(filesConfig.throttleDefinitions(), config),
+                        adminConfig.upgradeThrottlesFile(),
+                        SystemSetup::parseThrottles),
+                new AutoSysFileUpdate(
+                        createFileID(filesConfig.networkProperties(), config),
+                        adminConfig.upgradePropertyOverridesFile(),
+                        in -> parseConfig("override network properties", in)),
+                new AutoSysFileUpdate(
+                        createFileID(filesConfig.hapiPermissions(), config),
+                        adminConfig.upgradePermissionOverridesFile(),
+                        in -> parseConfig("override HAPI permissions", in)));
+        autoUpdates.forEach(update -> {
+            if (update.tryIfPresent(adminConfig.upgradeSysFilesLoc(), systemContext)) {
+                dispatch.stack().commitFullStack();
+            }
+        });
+    }
+
+    /**
+     * Encapsulates logic to attempt the automatic update of a system file by parsing the new contents of the
+     * corresponding file from the post-upgrade location.
+     *
+     * @param fileId the file ID of the system file
+     * @param upgradeFileName the name of the upgrade file
+     * @param upgradeFileParser the function to parse the upgrade file
+     */
+    private record AutoSysFileUpdate(
+            @NonNull FileID fileId,
+            @NonNull String upgradeFileName,
+            @NonNull Function<InputStream, Bytes> upgradeFileParser) {
+        /**
+         * Attempts to update the system file using the given system context if the corresponding upgrade file is
+         * present at the given location and can be parsed with this update's parser.
+         *
+         * @param postUpgradeLoc the location of the post-upgrade files
+         * @param systemContext the system context
+         * @return {@code true} if a synthetic update was dispatched, {@code false} otherwise
+         */
+        boolean tryIfPresent(@NonNull final String postUpgradeLoc, @NonNull final SystemContext systemContext) {
+            final var path = Paths.get(postUpgradeLoc, upgradeFileName);
+            if (!Files.exists(path)) {
+                log.info(
+                        "No post-upgrade file for {} found at {}, not updating",
+                        upgradeFileName,
+                        path.toAbsolutePath());
+                return false;
+            }
+            try (final var fin = Files.newInputStream(path)) {
+                final Bytes bytes;
+                try {
+                    bytes = upgradeFileParser.apply(fin);
+                } catch (Exception e) {
+                    log.warn("Failed to parse upgrade file for {} at {}", upgradeFileName, path.toAbsolutePath(), e);
```

Comment:
Failure to parse seems kind of serious.  It really means that a _bad_ file was stuck into the upgrade package, an operational error that should never occur (but, people being people, might just happen).  Anyway, given that the presence of the file in the package would indicate that _some_ kind of change was desired and that at least the `*.properties` files might affect the behavior of the network in important ways, should the system continue to come up if any of these files can't be parsed?  Perhaps starting the network would cause some transactions to be processed incorrectly (wrong properties applied).  It could even be that starting the network would cause some migration to be applied when it shouldn't be.

Reply:
True @david-bakin-sl , but a failure here can be remediated by manually submitting a `FileUpdate` to the network post-upgrade.

So we wouldn't want to fail _hard_ here.

Reply:
ok Michael! Just wanted to make sure that someone who knew more about node operations than me had/would consider it. TY!

(But it does mean that the node upgrade recipe needs to include a scan of the logs for WARN - or at least this specific message - before going live, if that isn't already done, so that the operators know they need to do the file update.  Is it?  WARN isn't a very high log level.)

Reply:
True, promoted these to `ERROR`. 💯